src/tls_stream.rs

```diff
@@ -501,6 +502,30 @@ impl<S> TlsStream<S>
         };
 
         unsafe {
+            // check if we trust the root-CA explicitly
+            let mut para_flags = winapi::CERT_CHAIN_POLICY_IGNORE_ALL_REV_UNKNOWN_FLAGS;
+            if let Some(ref mut store) = self.cert_store {
+                let cert_chain = *cert_chain.0;
+                if cert_chain.cChain > 0 {
+                    let first_rgp_chain = **cert_chain.rgpChain;
```

Comment:
I think we want to be looking at the final chain according to the docs: https://msdn.microsoft.com/en-us/library/windows/desktop/aa377182(v=vs.85).aspx (see the rgpChain section).